test/puppetlabs/puppetdb/cli/shovel_test.clj

```diff
@@ -0,0 +1,141 @@
+(ns puppetlabs.puppetdb.cli.shovel-test
+  (:import [java.nio.file Files])
+  (:require [clojure.test :refer :all]
+            [puppetlabs.puppetdb.cli.shovel :refer :all]
+            [puppetlabs.puppetdb.config :as conf]
+            [puppetlabs.puppetdb.mq :as mq]
+            [puppetlabs.puppetdb.testutils.services :as svc-utils]
+            [puppetlabs.puppetdb.examples :as ex]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.testutils.cli :as cli-utils]
+            [puppetlabs.puppetdb.cli.services :refer [shared-globals]]
+            [puppetlabs.puppetdb.scf.storage-utils :as sutils]
+            [puppetlabs.puppetdb.testutils.db :refer [*db* with-test-db]]
+            [puppetlabs.puppetdb.testutils :as tutils]
+            [puppetlabs.trapperkeeper.app :refer [get-service]]
+            [puppetlabs.trapperkeeper.services :refer [service-context]]
+            [puppetlabs.kitchensink.core :as ks]
+            [metrics.counters :as mc]
+            [puppetlabs.trapperkeeper.testutils.logging :refer [with-log-suppressed-unless-notable]]
+            [puppetlabs.puppetdb.testutils.log :as tlog]))
+
+(defn enqueue-and-shutdown [config commands]
+  (let [broker (mq/build-and-start-broker! "localhost"
+                                           (mq-dir config)
+                                           (:command-processing config))
+        endpoint (conf/mq-endpoint config)]
+
+    (.setPersistent broker true)
+    (try
+      (mq/start-broker! broker)
+
+      (with-open [factory (mq/activemq-connection-factory (conf/mq-broker-url config))
+                  conn (doto (.createConnection factory) .start)]
+        (doseq [[header command] commands]
+          (mq/send-message! conn
+                            endpoint
+                            (json/generate-string command)
+                            header)))
+      (finally
+        (mq/stop-broker! broker)))))
+
+(defn dlo-context [server]
+  (-> (get-service server :PuppetDBServer)
+      shared-globals
+      :dlo))
+
+(defn dlo-paths [server]
+  (with-open [path-stream (-> server
+                              dlo-context
+                              :path
+                              Files/newDirectoryStream)]
+    (mapv (comp str #(.toAbsolutePath %))
+          (-> path-stream .iterator iterator-seq))))
+
+(deftest shovel-upgrade-test
```

Comment:
Do we need more tests here? I'm testing a happy path upgrade, an upgrade with a "bad message" and a "no upgrade" scenario.